sympy/core/expr.py

```diff
@@ -826,7 +827,11 @@ def _eval_interval(self, x, a, b):
         value = B - A
 
         if a.is_comparable and b.is_comparable:
-            singularities = list(solveset(self.cancel().as_numer_denom()[1], x))
+            if a < b:
```

Comment:
Though it would have looked nice having the use "shortcut conditional expression", i.e `domain = Interval(a, b) if a > b else Interval(b, a)`.